### PR TITLE
fix: buy now && collectible loading in the container

### PR DESCRIPTION
**Pull request by BillySh1**
closes # MASK-500 MASK-411

### CODE REVIEW

File: packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx [hunk 1]
```diff
@@ -270,11 +276,19 @@ export function Collectible(props: CollectibleProps) {
...
+                            <div className={classes.loading}>
+                                <LoadingAnimation />
+                            </div>
+                        )) || (
+                            <div>
+                                {tabIndex === CollectibleTab.ARTICLE ? <ArticleTab /> : null}
+                                {tabIndex === CollectibleTab.TOKEN ? <TokenTab /> : null}
+                                {tabIndex === CollectibleTab.OFFER ? <OfferTab /> : null}
+                                {tabIndex === CollectibleTab.LISTING ? <ListingTab /> : null}
+                                {tabIndex === CollectibleTab.HISTORY ? <HistoryTab /> : null}
```
Comment (septs) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx:289]: plz extract the logic as component

```tsx
const CollectibleWindowRoute: FC<{ tabIndex: ?type? }> = ({ tabIndex }) => {
  if (tabIndex === CollectibleTab.ARTICLE) return <ArticleTab />
  if (tabIndex === CollectibleTab.TOKEN) return <TokenTab />
  // more route item
  return null
}
```

**Comment by orange187** [comment]
Checked

File: packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx (hunk 1 above)
Comment (BillySh1) [reply packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx]: OKK DONE

File: packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx [hunk 2]
```diff
@@ -195,6 +195,18 @@ export function Collectible(props: CollectibleProps) {
         <Tab className={classes.tab} key="history" label={t('plugin_collectible_history')} />,
     ]
 
+    const CollectibleTabsRouter: FC<{ tabIndex: number }> = ({ tabIndex }) => {
+        const tabMap: { [index: number]: ReactElement } = {
```
Comment (septs) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx:199]:
```suggestion
        const tabMap: Record<number, ReactElement> = {
```

**Comment by orange187** [comment]
Checked

File: packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx [hunk 3]
```diff
@@ -270,11 +288,11 @@ export function Collectible(props: CollectibleProps) {
...
-                        {tabIndex === CollectibleTab.ARTICLE ? <ArticleTab /> : null}
-                        {tabIndex === CollectibleTab.TOKEN ? <TokenTab /> : null}
-                        {tabIndex === CollectibleTab.OFFER ? <OfferTab /> : null}
-                        {tabIndex === CollectibleTab.LISTING ? <ListingTab /> : null}
-                        {tabIndex === CollectibleTab.HISTORY ? <HistoryTab /> : null}
+                        {(asset.loading && (
+                            <div className={classes.loading}>
+                                <LoadingAnimation />
+                            </div>
+                        )) || <div>{CollectibleTabsRouter({ tabIndex })}</div>}
```
Comment (septs) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx:295]:
```suggestion
                        )) || <div>{renderTab(tabIndex)}</div>}
```

File: packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx [hunk 4]
```diff
@@ -189,6 +195,18 @@ export function Collectible(props: CollectibleProps) {
         <Tab className={classes.tab} key="history" label={t('plugin_collectible_history')} />,
     ]
 
+    const CollectibleTabsRouter: FC<{ tabIndex: number }> = ({ tabIndex }) => {
```
Comment (septs) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx:198]:
```suggestion
    const renderTab = (tabIndex: number) => {
```

File: packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx [hunk 5]
```diff
@@ -270,11 +288,11 @@ export function Collectible(props: CollectibleProps) {
...
-                        {tabIndex === CollectibleTab.ARTICLE ? <ArticleTab /> : null}
-                        {tabIndex === CollectibleTab.TOKEN ? <TokenTab /> : null}
-                        {tabIndex === CollectibleTab.OFFER ? <OfferTab /> : null}
-                        {tabIndex === CollectibleTab.LISTING ? <ListingTab /> : null}
-                        {tabIndex === CollectibleTab.HISTORY ? <HistoryTab /> : null}
+                        {(asset.loading && (
+                            <div className={classes.loading}>
+                                <LoadingAnimation />
+                            </div>
+                        )) || <div>{renderTab(tabIndex)}</div>}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx:295]:
```suggestion
                        )) || <>{renderTab(tabIndex)}</>}
```

File: packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx [hunk 6]
```diff
@@ -189,6 +195,18 @@ export function Collectible(props: CollectibleProps) {
...
+        const tabMap: Record<number, ReactElement> = {
+            [CollectibleTab.ARTICLE]: <ArticleTab />,
+            [CollectibleTab.TOKEN]: <TokenTab />,
+            [CollectibleTab.OFFER]: <OfferTab />,
+            [CollectibleTab.LISTING]: <ListingTab />,
+            [CollectibleTab.HISTORY]: <HistoryTab />,
+        }
+
+        return tabMap[tabIndex] || null
+    }
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx:208]:
```suggestion
    const renderTab = (tabIndex: CollectibleTab) => {
        const tabMap: Record<CollectibleTab, ReactElement> = {
            [CollectibleTab.ARTICLE]: <ArticleTab />,
            [CollectibleTab.TOKEN]: <TokenTab />,
            [CollectibleTab.OFFER]: <OfferTab />,
            [CollectibleTab.LISTING]: <ListingTab />,
            [CollectibleTab.HISTORY]: <HistoryTab />,
        }

        return tabMap[tabIndex] || null
    }
```

File: packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx (hunk 5 above)
Comment (BillySh1) [reply packages/mask/src/plugins/Collectible/SNSAdaptor/Collectible.tsx]: committed

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**